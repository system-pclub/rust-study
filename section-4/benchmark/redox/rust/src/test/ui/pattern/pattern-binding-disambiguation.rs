struct UnitStruct;
struct TupleStruct();
struct BracedStruct{}

enum E {
    UnitVariant,
    TupleVariant(),
    BracedVariant{},
}
use E::*;

const CONST: () = ();
static STATIC: () = ();

fn function() {}

fn main() {
    let doesnt_matter = 0;

    match UnitStruct {
        UnitStruct => {} // OK, `UnitStruct` is a unit struct pattern
    }
    match doesnt_matter {
        TupleStruct => {} //~ ERROR match bindings cannot shadow tuple structs
    }
    match doesnt_matter {
        BracedStruct => {} // OK, `BracedStruct` is a fresh binding
    }
    match UnitVariant {
        UnitVariant => {} // OK, `UnitVariant` is a unit variant pattern
    }
    match doesnt_matter {
        TupleVariant => {} //~ ERROR match bindings cannot shadow tuple variants
    }
    match doesnt_matter {
        BracedVariant => {} //~ ERROR match bindings cannot shadow struct variants
    }
    match CONST {
        CONST => {} // OK, `CONST` is a const pattern
    }
    match doesnt_matter {
        STATIC => {} //~ ERROR match bindings cannot shadow statics
    }
    match doesnt_matter {
        function => {} // OK, `function` is a fresh binding
    }

    let UnitStruct = UnitStruct; // OK, `UnitStruct` is a unit struct pattern
    let TupleStruct = doesnt_matter; //~ ERROR let bindings cannot shadow tuple structs
    let BracedStruct = doesnt_matter; // OK, `BracedStruct` is a fresh binding
    let UnitVariant = UnitVariant; // OK, `UnitVariant` is a unit variant pattern
    let TupleVariant = doesnt_matter; //~ ERROR let bindings cannot shadow tuple variants
    let BracedVariant = doesnt_matter; //~ ERROR let bindings cannot shadow struct variants
    let CONST = CONST; // OK, `CONST` is a const pattern
    let STATIC = doesnt_matter; //~ ERROR let bindings cannot shadow statics
    let function = doesnt_matter; // OK, `function` is a fresh binding
}
