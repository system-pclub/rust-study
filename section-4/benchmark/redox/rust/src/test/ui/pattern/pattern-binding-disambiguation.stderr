error[E0530]: match bindings cannot shadow tuple structs
  --> $DIR/pattern-binding-disambiguation.rs:24:9
   |
LL | struct TupleStruct();
   | --------------------- the tuple struct `TupleStruct` is defined here
...
LL |         TupleStruct => {}
   |         ^^^^^^^^^^^ cannot be named the same as a tuple struct

error[E0530]: match bindings cannot shadow tuple variants
  --> $DIR/pattern-binding-disambiguation.rs:33:9
   |
LL | use E::*;
   |     ---- the tuple variant `TupleVariant` is imported here
...
LL |         TupleVariant => {}
   |         ^^^^^^^^^^^^ cannot be named the same as a tuple variant

error[E0530]: match bindings cannot shadow struct variants
  --> $DIR/pattern-binding-disambiguation.rs:36:9
   |
LL | use E::*;
   |     ---- the struct variant `BracedVariant` is imported here
...
LL |         BracedVariant => {}
   |         ^^^^^^^^^^^^^ cannot be named the same as a struct variant

error[E0530]: match bindings cannot shadow statics
  --> $DIR/pattern-binding-disambiguation.rs:42:9
   |
LL | static STATIC: () = ();
   | ----------------------- the static `STATIC` is defined here
...
LL |         STATIC => {}
   |         ^^^^^^ cannot be named the same as a static

error[E0530]: let bindings cannot shadow tuple structs
  --> $DIR/pattern-binding-disambiguation.rs:49:9
   |
LL | struct TupleStruct();
   | --------------------- the tuple struct `TupleStruct` is defined here
...
LL |     let TupleStruct = doesnt_matter;
   |         ^^^^^^^^^^^ cannot be named the same as a tuple struct

error[E0530]: let bindings cannot shadow tuple variants
  --> $DIR/pattern-binding-disambiguation.rs:52:9
   |
LL | use E::*;
   |     ---- the tuple variant `TupleVariant` is imported here
...
LL |     let TupleVariant = doesnt_matter;
   |         ^^^^^^^^^^^^ cannot be named the same as a tuple variant

error[E0530]: let bindings cannot shadow struct variants
  --> $DIR/pattern-binding-disambiguation.rs:53:9
   |
LL | use E::*;
   |     ---- the struct variant `BracedVariant` is imported here
...
LL |     let BracedVariant = doesnt_matter;
   |         ^^^^^^^^^^^^^ cannot be named the same as a struct variant

error[E0530]: let bindings cannot shadow statics
  --> $DIR/pattern-binding-disambiguation.rs:55:9
   |
LL | static STATIC: () = ();
   | ----------------------- the static `STATIC` is defined here
...
LL |     let STATIC = doesnt_matter;
   |         ^^^^^^ cannot be named the same as a static

error: aborting due to 8 previous errors

For more information about this error, try `rustc --explain E0530`.
